sbi/inference/base.py

```diff
@@ -116,6 +116,14 @@ def __init__(
                 each round.
         """
 
+        # We set the device globally by setting the default tensor type for all tensors.
+        assert device in (
+            "gpu",
+            "cpu",
+        ), "Currently, only 'gpu' or 'cpu' are supported as devices."
+        set_default_device(device)
+        self._device = get_default_device()
```

Comment:
How about we merge these two into a single function? E.g. `get_default_device` takes `device` as input argument.